opm/io/eclipse/rst/segment.cpp

```diff
@@ -33,6 +33,23 @@ double area_to_si(const Opm::UnitSystem& unit_system, const double raw_value)
     return unit_system.to_si(M::length, unit_system.to_si(M::length, raw_value));
 }
 
+double
+load_device_base_strength(const Opm::UnitSystem& unit_system,
+                          const int              segment_type,
+                          const double           base_strength_raw)
+{
+    auto unit = Opm::UnitSystem::measure::identity;
+
+    if (segment_type == -7) {   // SICD
```

Comment:
less magic?

Reply:
> less magic?

Yeah, good point.  In general, I wanted to make `RstSegment` independent of [`Segment`](https://github.com/OPM/opm-common/blob/0822f8b4c0167f630194db002301a4d21304a6c1/opm/input/eclipse/Schedule/MSW/Segment.cpp#L47), but I think the **real** fix is to add named constants for these values in `VectorItems/msw.hpp`.  I'll make an update to that effect.

Reply:
> > less magic?
> 
> Yeah, good point. In general, I wanted to make `RstSegment` independent of [`Segment`](https://github.com/OPM/opm-common/blob/0822f8b4c0167f630194db002301a4d21304a6c1/opm/input/eclipse/Schedule/MSW/Segment.cpp#L47), but I think the **real** fix is to add named constants for these values in `VectorItems/msw.hpp`. I'll make an update to that effect.

I've pushed that update.  You're welcome to have a look at the new version of the PR to consider the level of magic now.

Reply:
houdini is sad. looks much better. :+1:

Reply:
> houdini is sad. looks much better. 👍

Thanks.  There is one (expected) [regression failure](https://ci.opm-project.org/job/opm-common-PR-builder/7207/testReport/) from this work and that will require updating the reference solutions.  I will do that later, once we've incorporated the other reference solution updates in flight.